clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1107,7 +1067,7 @@ public Map<TopicPartition, OffsetAndTimestamp> offsetsForTimes(Map<TopicPartitio
                 return listOffsetsEvent.emptyResults();
             }
 
-            return applicationEventHandler.addAndGet(listOffsetsEvent, timer)
+            return applicationEventHandler.addAndGet(listOffsetsEvent)
```

Comment:
The timer is only used by the event. Maybe a deadline is simpler.

Reply:
Removed use of `Timer` in favor of calculating the deadline from the `time` and `timeout` directly.